### PR TITLE
feat(equals): Add equals implementation

### PR DESCRIPTION
**Pull request by RockoDev**

### CODE REVIEW

**Comment by RockoDev** [comment]
This only works with primitive values. I don't know if it's worth adding support for other types of values.

Also, this function compares an "unlimited" number of arrays:
```javascript
function equals(...arrs) {
  for (let i1 in arrs) {
    if (i1===0) continue;
    if (arrs[0].length !== arrs[i1].length) return false;
    for (let i2 in arrs[i1]) {
      if (arrs[0][i2] !== arrs[i1][i2]) return false;
    }
  }
  return true;
}
```
Perhaps it's not necessary to make it more complex.

What do you think?

File: f/equalsAsync.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+async function equals(it1, it2) {
+  it1 = it1[Symbol.iterator]();
+  it2 = it2[Symbol.iterator]();
+  for await (;;) {
```
Comment (surma) [inline f/equalsAsync.js]: Sadly, this is not how `for await` works 😅 The only implicit await construct is `for await(let x of asyncIterator) { ... }`.

In this case, you’d probably use a normal `while(true)` and say `let i1 = await it1.next(), ...`.

File: test/equals.test.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+it('returns true if both iterables have equal length and values (in the same order)', function () {
+  expect(f([1, 2, 3, 'hello'], [1, 2, 3, 'hello'])).to.deep.equal(true);
```
Comment (surma) [inline test/equals.test.js]: I’d remove the `deep` here.

**Comment by surma** [comment]
Meeeeerged! 👏 Thanks @RockoDev

**Pull request closed without merging**